MFVNeutralino/plugins/Vertexer.cc

```diff
@@ -1,891 +1,1774 @@
+            printf("\n");
+          }
+
+          *v[0] = vnm1;
+          ++refit_count[iv];
+          --v[0], --iv;
+          break;
+        }
+      }
+    }
+  }
+
+  if (histos_output_afterdzfit){
+    fillCommonOutputHists(vertices, fake_bs_vtx, tt_builder, stepEnum::afterdzfit);
+  }
+
+
+  /////////////////////////////////////////////////////////////////////////////////////////////////////
+  // Merge every pair of output vertices that satisfy the following criteria to resolve split-vertices:
+  //   - >=2trk/vtx
+  //   - dBV > 100 um
+  //   - |dPhi(vtx0,vtx1)| < 0.5 
+  //   - svdist2d < 300 um
+  // Note that the merged vertex must pass chi2/dof < 5
+  ////////////////////////////////////////////////////////////////////////////////////////////////////
+  if (resolve_split_vertices_tight) {
+    int merge_pair_count = 0;
+    reco::VertexCollection potential_merged_vertices;
+
+
+    for (v[0] = vertices->begin(); v[0] != vertices->end(); ++v[0]) {
+
+      track_set tracks[2];
+      tracks[0] = vertex_track_set(*v[0]);
+
+      bool merge = false;
+      for (v[1] = v[0] + 1; v[1] != vertices->end(); ++v[1]) {
+        if (vertices->size() >= 2 && v[0]->nTracks() >= 2 && v[1]->nTracks() >= 2) {
+
+          tracks[1] = vertex_track_set(*v[1]);
+
+          Measurement1D v_dist = vertex_dist_2d.distance(*v[0], *v[1]);
+
+          Measurement1D dBV0_Meas1D = vertex_dist_2d.distance(*v[0], fake_bs_vtx);
+          double dBV0 = dBV0_Meas1D.value();
+
+          Measurement1D dBV1_Meas1D = vertex_dist_2d.distance(*v[1], fake_bs_vtx);
+          double dBV1 = dBV1_Meas1D.value();
+
+          double v0x = v[0]->x() - bsx;
+          double v0y = v[0]->y() - bsy;
+
+          double phi0 = atan2(v0y, v0x);
+
+          double v1x = v[1]->x() - bsx;
+          double v1y = v[1]->y() - bsy;
+
+          double phi1 = atan2(v1y, v1x);
+
+          if (fabs(reco::deltaPhi(phi0, phi1)) < 0.5 && v_dist.value() < 0.0300 && dBV0 > 0.0100 && dBV1 > 0.0100) {
+            track_set tracks_to_fit;
+            for (int i = 0; i < 2; ++i)
+              for (auto tk : tracks[i])
+                tracks_to_fit.insert(tk);
+            std::vector<reco::TransientTrack> ttks;
+            for (auto tk : tracks_to_fit)
+              ttks.push_back(seed_tracks[seed_track_ref_map[tk]]);
+
+            if (investigate_merged_vertices) {
+              std::vector<TransientVertex> tv(1, kv_reco->vertex(ttks));
+              potential_merged_vertices.push_back(reco::Vertex(tv[0]));
+              //std::cout << "ntrack in potental merged: " << potential_merged_vertices.back().nTracks() << std::endl;
+            }
+            reco::VertexCollection merged_vertices;
+            for (const TransientVertex& tv : kv_reco_dropin(ttks))
+              merged_vertices.push_back(reco::Vertex(tv));
+
+            if (merged_vertices.size() == 1 && vertex_track_set(merged_vertices[0], 0) == tracks_to_fit) {
+
+              merge_pair_count += 1;

```

Comment:
We don't actually use this anywhere, but maybe it would be useful as part of the verbose info? Or we can drop it if you don't think it would be useful.

Reply:
Yes, I think we can drop it. The same info can be deduced from the change in entries of ver-level plots `afterdzfit` and `aftermerge`.